### PR TITLE
Support non-encoded `filePath` with assumption that `filePath` appears as the last parameter.

### PR DESCRIPTION
**Pull request by junhaoliao**
# References
Customer wishes to input non-encoded `filePath` as part of the browser client URL. 

Our analysis shows that the non-encoded value of `filePath` should appear as the last parameter in the browser client URL, so that Log Viewer can assume all characters after `filePath=` and before `#` are the value of `filePath`. With the proposed URL schema, any `&<paramter_name>=<parameter_value>` parsed after `filePath` are assumed to be part of the `filePath` value. 

# Description
1. Add parser handling to get `filePath` as the last parameter in the browser client URL.
2. Add export handling to set `filePath` as the last parameter in the browser client URL. 
3. Show only file name rather than `filePath`'s last segment in MenuBar. 

# Validation performed
1. Visited `http://localhost:3010/?filePath=https://yscope.s3.us-east-2.amazonaws.com/sample-logs/yarn-ubuntu-resourcemanager-ip-172-31-17-135.log.1.clp.zst#logEventIdx=375558` and observed the log file to be loaded and decoded properly.
2. Opened the browser's debugger and switched to "Network" tab in the debugger. Visited `http://localhost:3010/?filePath=https://yscope.s3.us-east-2.amazonaws.com/sample-logs/yarn-ubuntu-resourcemanager-ip-172-31-17-135.log.1.clp.zst?user=user&pass=pass#logEventIdx=375558` and observed the log file to be loaded and decoded properly. Observed in the debugger a `GET` request was made with `Request URL` being `https://yscope.s3.us-east-2.amazonaws.com/sample-logs/yarn-ubuntu-resourcemanager-ip-172-31-17-135.log.1.clp.zst?user=user&pass=pass`.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -53,15 +53,17 @@ export function App () {
      * If neither are provided, we display a prompt to load a file.
      */
     const init = () => {
-        const urlSearchParams = new URLSearchParams(window.location.search.substring(1));
+        const urlSearchParams = new URLSearchParams(window.location.search);
```
Comment (junhaoliao) [inline src/App.js]: It appears that URLSearchParams() can correctly parse a string starting with a "?" but not a "#", so we do not have to cut the "?".

Comment (kirkrodrigues) [reply src/App.js]: Hm, I wonder if it's better to just stay consistent and use substring(1) for both.

File: src/Viewer/services/decoder/FileManager.js [hunk 2]
```diff
@@ -738,7 +747,7 @@ class FileManager {
         searchChunk(0);
     };
 
-   /**
+    /**
```
Comment (kirkrodrigues) [inline src/Viewer/services/decoder/FileManager.js]: Undo

File: src/Viewer/components/LeftPanel/DownloadHelper.js [hunk 3]
```diff
@@ -31,8 +31,8 @@ const downloadCompressedFile = () => {
     //  which avoids interruption of uncompressed logs download
     link.target = "_blank";
 
-    const urlParams = new URLSearchParams(window.location.search);
-    const filePath = urlParams.get("filePath");
+    let filePath = window.location.search.split("filePath=")[1];
```
Comment (kirkrodrigues) [inline src/Viewer/components/LeftPanel/DownloadHelper.js]: Seems like we're using this logic in three places. Can we extract it into a method?

File: src/Viewer/services/utils.js [hunk 4]
```diff
@@ -54,11 +54,20 @@ const modifyPage = (action, currentPage, requestedPage, pages) => {
 const getModifiedUrl = (searchParams, hashParams) => {
     const url = new URL(`${window.location.origin}${window.location.pathname}`);
 
-    const urlSearchParams = new URLSearchParams(window.location.search.substring(1));
+    let filePath = window.location.search.split("filePath=")[1];
+    filePath = (undefined === filePath) ? null : filePath.substring(filePath.indexOf("#"));
+
+    const splitLocationSearch = window.location.search.split(`filePath=${filePath}`);
+    const locationSearchWithoutFilePath = splitLocationSearch.join("");
```
Comment (kirkrodrigues) [inline src/Viewer/services/utils.js]:
```suggestion
    const locationSearchWithoutFilePath = window.location.search.split(`filePath=${filePath}`).join("");
```

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (junhaoliao) [reply src/App.js]: Sure, since adding .substring(1) is harmless.

**Review by kirkrodrigues — APPROVED** [review]

**Pull request merged**